server/src/tests/table_api.rs

```diff
@@ -0,0 +1,153 @@
+            IdentifierWrapper(ident_str!("String").into()),
+            Vec::new(),
+        ));
+        let request = TableItemRequest {
+            key_type: MoveType::Vector {
+                items: Box::new(string_type.clone()),
+            },
+            value_type: MoveType::Vector {
+                items: Box::new(string_type.clone()),
+            },
+            key: serde_json::Value::Array(vec![
+                serde_json::Value::String("abc".into()),
+                serde_json::Value::String("abc".into()),
+            ]),
+        };
+        let table_item = ctx
+            .mv_get_table_item(&handle, request.clone(), BlockNumberOrTag::Latest)
+            .await
+            .unwrap();
+
+        // The key and value are equal for this table
+        assert_eq!(table_item, request.key);
+
+        ctx.shutdown().await;
+        Ok(())
+    })
+    .await
+}
+
+async fn call_tables_contract(
+    ctx: &mut TestContext<'static>,
+    signer: &PrivateKeySigner,
+) -> TransactionReceipt {
+    let address = signer.address();
+    let input = TransactionData::EntryFunction(EntryFunction::new(
+        ModuleId::new(address.to_move_address(), ident_str!("tables").into()),
+        ident_str!("make_test_tables").into(),
+        Vec::new(),
+        vec![bcs::to_bytes(&MoveValue::Signer(address.to_move_address())).unwrap()],
+    ));
+    let mut tx = TxEip1559 {
+        chain_id: ctx.genesis_config.chain_id,
+        nonce: 1,
+        gas_limit: u64::MAX,
+        max_fee_per_gas: 0,
+        max_priority_fee_per_gas: 0,
+        to: TxKind::Call(address),
+        value: U256::ZERO,
+        access_list: Default::default(),
+        input: input.to_bytes().unwrap().into(),
+    };
+    let signature = signer.sign_transaction_sync(&mut tx).unwrap();
+    let signed_tx = TxEnvelope::Eip1559(tx.into_signed(signature));
+    ctx.execute_transaction(signed_tx).await.unwrap()
+}
+
+async fn deploy_tables_contract(
+    ctx: &mut TestContext<'static>,
+    signer: &PrivateKeySigner,
+) -> TransactionReceipt {
+    let address = signer.address();
+    let bytecode = create_move_tables_contract_bytecode(address);
+    let mut tx = TxEip1559 {
+        chain_id: ctx.genesis_config.chain_id,
+        nonce: 0,
+        gas_limit: u64::MAX,
+        max_fee_per_gas: 0,
+        max_priority_fee_per_gas: 0,
+        to: TxKind::Create,
+        value: U256::ZERO,
+        access_list: Default::default(),
+        input: bytecode.into(),
+    };
+    let signature = signer.sign_transaction_sync(&mut tx).unwrap();
+    let signed_tx = TxEnvelope::Eip1559(tx.into_signed(signature));
+    ctx.execute_transaction(signed_tx).await.unwrap()
+}
+
+fn create_move_tables_contract_bytecode(address: Address) -> Vec<u8> {
+    let bytecode_hex = std::fs::read_to_string("src/tests/res/tables.hex").unwrap();
+    let bytecode = hex::decode(bytecode_hex.trim()).unwrap();
+    crate::tests::integration::set_module_address(bytecode, address)
+}
```

Comment:
Can we have some short description detailing actual contract code? E.g. what functions it has and such, or a link from where it was taken if it's Aptos standard something.